### PR TITLE
domain: rename `binary_path` to `deploy_path`

### PR DESCRIPTION
**Pull request by mapleFU**
Signed-off-by: mapleFU <1506118561@qq.com>

### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary:

In tidb cluster, all components has a content like `binary_path` or `deploy_path`. And we want to unify them to `deploy_path`.

### What is changed and how it works?

What's Changed:

field name for `binary_path`.

### Check List 

Tests 

- No code

Side effects

No

### Release note 

* Unify the component interface and change `binary_path` to `deploy_path`.

### CODE REVIEW

File: domain/infosync/info.go [hunk 1]
```diff
@@ -316,7 +316,7 @@ func (is *InfoSyncer) getTopologyInfo() topologyInfo {
 			GitHash: is.info.ServerVersionInfo.GitHash,
 		},
 		StatusPort:     is.info.StatusPort,
-		BinaryPath:     s,
+		DeployPath:     s,
```
Comment (breeswish) [inline domain/infosync/info.go:319]: what about places that assigning values?

**Comment by breeswish** [comment]
LGTM

**Comment by zz-jason** [comment]
@mapleFU please write a release note.

**Comment by mapleFU** [comment]
@zz-jason added.

**Comment by zz-jason** [comment]
@mapleFU do we need to cherry-pick this PR to release branches?

**Comment by mapleFU** [comment]
@zz-jason we should cherry-pick it to 4.0

**Pull request closed without merging**